Unable to start local development using Docker Compose 2.19.0

### What went wrong?

**What happened**:
  - `make start` fails with an error: `service postgres is required by grafana but is disabled. Can be enabled by profiles [postgres]`

**What did you expect to happen**:
  - starting development containers


### How do we reproduce it?

1. Use docker-compose v2.19.1
2. Use `COMPOSE_PROFILES=oncall_ui,redis,grafana,engine`
3. `make init`
4. `make start`

```
COMPOSE_PROFILES=oncall_ui,redis,grafana,engine DB=sqlite BROKER_TYPE=redis SQLITE_DB_FILE=./engine/oncall.db docker compose -f docker-compose-developer.yml up --remove-orphans -d
service postgres is required by grafana but is disabled. Can be enabled by profiles [postgres]
make: *** [start] Error 1
```


### Grafana OnCall Version

v1.3.1

### Product Area

Other

### Grafana OnCall Platform?

Docker

### User's Browser?

_No response_

### Anything else to add?

I suppose it is caused by [this fix](https://github.com/docker/compose/pull/10602) and now `depends_on` is required by default.

As I understand `required` field would be implemented introduced in https://github.com/compose-spec/compose-spec/pull/382 

But now using docker compose v2.18.1 is the only option.

--- a/dev/README.md
+++ b/dev/README.md
@@ -35,7 +35,7 @@ environment variable.
    **NOTE**: the `docker-compose-developer.yml` file uses some syntax/features that are only supported by Docker Compose
    v2. For instructions on how to enable this (if you haven't already done so),
    see [here](https://www.docker.com/blog/announcing-compose-v2-general-availability/). Ensure you have Docker Compose
-   version 2.10 or above installed - update instructions are [here](https://docs.docker.com/compose/install/linux/).
+   version 2.20.2 or above installed - update instructions are [here](https://docs.docker.com/compose/install/linux/).
 2. Run `make init start`. By default this will run everything in Docker, using SQLite as the database and Redis as the
    message broker/cache. See [`COMPOSE_PROFILES`](#compose_profiles) below for more details on how to swap
    out/disable which components are run in Docker.

--- a/docker-compose-developer.yml
+++ b/docker-compose-developer.yml
@@ -144,12 +144,16 @@ services:
     depends_on:
       postgres:
         condition: service_healthy
+        required: false
       mysql:
         condition: service_healthy
+        required: false
       rabbitmq:
         condition: service_healthy
+        required: false
       redis:
         condition: service_healthy
+        required: false
     profiles:
       - engine
 
@@ -208,7 +212,7 @@ services:
     container_name: mysql
     labels: *oncall-labels
     image: mysql:8.0.32
-    command: >
+    command: >-
       --default-authentication-plugin=mysql_native_password --character-set-server=utf8mb4
       --collation-server=utf8mb4_unicode_ci --max_connections=1024
     restart: always
@@ -236,7 +240,7 @@ services:
     container_name: mysql_to_create_grafana_db
     labels: *oncall-labels
     image: mysql:8.0.32
-    command: >
+    command: >-
       bash -c "mysql -h mysql -uroot -pempty
       -e 'CREATE DATABASE IF NOT EXISTS grafana CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci;'"
     depends_on:
@@ -276,7 +280,7 @@ services:
     container_name: postgres_to_create_grafana_db
     labels: *oncall-labels
     image: postgres:14.4
-    command: >
+    command: >-
       bash -c "PGPASSWORD=empty psql -U postgres
       -h postgres -tc \"SELECT 1 FROM pg_database WHERE datname = 'grafana'\" | grep -q 1 || PGPASSWORD=empty psql
       -U postgres -h postgres -c \"CREATE DATABASE grafana\""
@@ -326,8 +330,10 @@ services:
     depends_on:
       postgres:
         condition: service_healthy
+        required: false
       mysql:
         condition: service_healthy
+        required: false
     profiles:
       - grafana
 volumes:

--- a/docker-compose-mysql-rabbitmq.yml
+++ b/docker-compose-mysql-rabbitmq.yml
@@ -28,8 +28,7 @@ services:
     restart: always
     ports:
       - "8080:8080"
-    command: >
-      sh -c "uwsgi --ini uwsgi.ini"
+    command: sh -c "uwsgi --ini uwsgi.ini"
     environment: *oncall-environment
     depends_on:
       mysql:
@@ -68,7 +67,7 @@ services:
 
   mysql:
     image: mysql:8.0.32
-    command: >
+    command: >-
       --default-authentication-plugin=mysql_native_password
       --character-set-server=utf8mb4 --collation-server=utf8mb4_unicode_ci
     restart: always
@@ -123,7 +122,7 @@ services:
 
   mysql_to_create_grafana_db:
     image: mysql:8.0.32
-    command: >
+    command: >-
       bash -c "mysql -h ${MYSQL_HOST:-mysql} -uroot -p${MYSQL_PASSWORD:?err}
       -e 'CREATE DATABASE IF NOT EXISTS grafana CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci;'"
     depends_on:

--- a/docker-compose.yml
+++ b/docker-compose.yml
@@ -22,8 +22,7 @@ services:
     restart: always
     ports:
       - "8080:8080"
-    command: >
-      sh -c "uwsgi --ini uwsgi.ini"
+    command: sh -c "uwsgi --ini uwsgi.ini"
     environment: *oncall-environment
     volumes:
       - oncall_data:/var/lib/oncall

